third_party/kiva/cmake/compiler-flags.cmake

```
@@ -15,7 +15,9 @@ $<$<OR:$<CXX_COMPILER_ID:GNU>,$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleCl
   -Wall
   -Wextra
   -Wpedantic
-  -Wno-enum-constexpr-conversion
+  $<$<AND:$<CXX_COMPILER_ID:Clang>,$<VERSION_GREATER_EQUAL:$<CXX_COMPILER_VERSION>,16>>:
+    -Wno-enum-constexpr-conversion
+  >
```

Eventually AppleClang will catch up with clang 16, and we're back with a warning there, no?

```suggestion
  $<$<AND:$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>>,$<VERSION_GREATER_EQUAL:$<CXX_COMPILER_VERSION>,16>>:
    -Wno-enum-constexpr-conversion
  >
```

Generator expressions quickly become unbearable, don't they?


As a side note, I realized we don't set CMP0025 to NEW, so CXX_COMPILER_ID:Clang matches both Clang and AppleClang **at the moment**.  https://cmake.org/cmake/help/latest/policy/CMP0025.html

---

You are right, and it is a bit more complicated than that actually, because `AppleClang` version does not necessarily match upstream `clang` version, see this nice table here: https://en.wikipedia.org/wiki/Xcode#cite_note-llvm-swift_5.8.1_version-187

Basically Apple numbers `AppleClang` the same as the released `XCode` version, whatever the upstream `clang` version is - and apparently `AppleClang` 15 corresponds to `clang` 16. So we need different conditions here. Given that `CMAKE_CXX_COMPILER_ID` is used in many third-party `CMakeLists.txt` I'd rather not change `CMP0025` now, I'll fix that without a generator expression.

---

> AppleClang version does not necessarily match upstream clang version

I had forgotten about this... Of course they do not, that'd be too simple.

---

Done, in the end I sticked with generator expressions but decomposed it in (hopefully) digestable pieces.